Update README.MD

Arthil opened the pull request:
Add 2 diagrams, which explain the BBB boot and daft runs. Also add more troubleshooting information and fix and complete some other sections.

Signed-off-by: Christian da Costa <christian.da.costa@intel.com>

skuusela commented on README.MD:
```diff
@@ -65,6 +68,23 @@ DAFT is also split code wise with _testing_harness_ including the code BBB uses
 called AFT (Automatic Flasher Tester) and the program that the PC uses is DAFT.
 
+![](readme_pictures/BBB_booting.png)
+_BeagleBone Black booting with the PC Host._
```
Add additional newline after picture so the text lines up better.

skuusela replied:
Also there is some random black arrowhead in the lower left corner.

skuusela commented on README.MD:
```diff
@@ -79,6 +99,21 @@ strictly part of DAFT but rather DAFT includes tools and libraries to interact
+_Running daft._
+
+1. Start daft by executing the following command in the PC Host: `daft DUT PATH/TO/IMG.img`. Replace DUT with the desired DUT e.g. `minnowboard` or `joule` and `PATH/TO/IMG.img` with the image you want to flash. You can also run the command directly in BBB but it then it won't lock the DUTs. Running daft in BBB is done with the command `aft DUT PATH/TO/IMG.img`.In case you want to only flash or t ...
+2. DAFT then checks if the device is blacklisted in the lockfiles. If not, then it locks it so other simultaneous DAFT runs won't use the same DUT.
```
DAFT checks if there is any free device of that type so not blacklisted or locked. Or if you call DAFT with specific device like `joule1` then it will only check that specific lockfile if its locked or blacklisted.

Arthil replied:
Good point, I will change that too!

skuusela commented on README.MD:
```diff
@@ -79,6 +100,21 @@ strictly part of DAFT but rather DAFT includes tools and libraries to interact
+
+1. Start daft by executing the following command in the PC Host: `daft DUT PATH/TO/IMG.img`. Replace DUT with the desired DUT e.g. `minnowboard` or `joule` and `PATH/TO/IMG.img` with the image you want to flash. You can also run the command directly in BBB but it then it won't lock the DUTs. Running daft in BBB is done with the command `aft DUT PATH/TO/IMG.img`.In case you want to only flash or t ...
+2. DAFT then checks if the device is blacklisted in the lockfiles. If not, then it locks it so other simultaneous DAFT runs won't use the same DUT.
+3. BBB reboots the DUT and proceeds to start the support image in the DUT. The support image is needed so that the target image can be flashed to the SDcard or to another memory storage. BBB emulates itself as multiple components to enable automated and simple function:
```
SD card*

skuusela commented on README.MD:
```diff
@@ -79,6 +100,21 @@ strictly part of DAFT but rather DAFT includes tools and libraries to interact
+2. DAFT then checks if the device is blacklisted in the lockfiles. If not, then it locks it so other simultaneous DAFT runs won't use the same DUT.
+3. BBB reboots the DUT and proceeds to start the support image in the DUT. The support image is needed so that the target image can be flashed to the SDcard or to another memory storage. BBB emulates itself as multiple components to enable automated and simple function:
+   1. As a keyboard (to get through DUT bios).
+   2. As a usbdrive (so that the DUT can boot the support image).
```
USB drive*

skuusela commented on README.MD:
```diff
@@ -79,6 +100,21 @@ strictly part of DAFT but rather DAFT includes tools and libraries to interact
+3. BBB reboots the DUT and proceeds to start the support image in the DUT. The support image is needed so that the target image can be flashed to the SDcard or to another memory storage. BBB emulates itself as multiple components to enable automated and simple function:
+   1. As a keyboard (to get through DUT bios).
+   2. As a usbdrive (so that the DUT can boot the support image).
+   3. As a ethernet (to give the DUT an IPaddress).
```
IP address* You could also generalize that ethernet is needed so DUT can be connected to as the ethernet is needed so BBB can run commands on it (needed for flashing and testing) and also so the DUT gets the image to flash.

skuusela commented on README.MD:
```diff
@@ -65,6 +68,24 @@ DAFT is also split code wise with _testing_harness_ including the code BBB uses
+2. Get an IP adress for the BBB from the DHCP server. Refer to [DHCP problems](#62-dhcp-problems) if you are having issues with this.
+3. Get the BBB bootsystem kernel via TFTP. The BBB will fetch the bootsystem kernel from the PC Host via TFTP. Refer to [TFTP problems](#63-tftp-problems) if there are problems.
+4. BBB executes the kernel, which is the bootsystem core and is needed to be able to boot the rest of the system via NFS.
+5. Boot the rest of the system via NFS. Refer to [NFS problems](#64-nfs-problems) if you have issues with NFS.
```
I think it's correct to say 'from NFS' instead of 'via NFS'.

Arthil replied:
via = by means of, so I don't see a problem here

skuusela commented on README.MD:
```diff
@@ -79,6 +100,21 @@ strictly part of DAFT but rather DAFT includes tools and libraries to interact
+   2. As a usbdrive (so that the DUT can boot the support image).
+   3. As a ethernet (to give the DUT an IPaddress).
+4. When the support image boots, it gets an IP address from the BBBs DHCP server (NOT THE PC HOSTS). If you want to, you can SSH to the DUT from the PC Host by first doing a SSH to the BBB and then another SSH to the DUT.
+5. Flash the target image fetched from the PC Host via NFS to the DUTs memory storage. The DUT doesn't connect directly to the PC Host but instead through the BBB, which forwards the connection both ways.
```
from the PC Host NFS*

skuusela commented on README.MD:
````diff
@@ -543,6 +580,8 @@ address:
 ```
 cat /var/lib/misc/dnsmasq.leases
 ```
+In some distros you don't have this file. In that case use `arp` in the terminal to do the same thing.
````
You always have the dnsmasq.leases file but the location might differ. Also `arp -n` is probably the command to go with.

skuusela commented on README.MD:
```diff
@@ -1053,6 +1099,25 @@ U-Boot is quite clunky with the pathnames so try different paths like
+
+### 6.2 DHCP problems
+If you are having problems with the DHCP server, you can try the following steps:
+1. Check that your dnsmasq service is on. In Fedora this is done by executing the following command `systemctl status dnsmasq`. If it is not, you need to start it with `systemctl start dnsmasq`. If it still isn't working, check that you have completed all the steps when configuring the [Dnsmasq](#232-dnsmasq).
```
You can drop the 'Fedora', most of the modern distros have systemd and these commands would be the same with all of them.

skuusela commented:
`Micro SD card for BBB U-Boot (optional)` The link on this line isn't working.

skuusela commented on README.MD:
```diff
@@ -202,7 +148,28 @@ the setup was also tested to work with Gigabyte NUC, Intel Edison, BeagleBone
+
+![](readme_pictures/BBB_booting.png)
+
+_BeagleBone Black booting with the PC Host._
```
The picture and the picture text would look nicer if the image didn't have so much white space at the bottom. Not a big deal though.

skuusela commented on README.MD:
```diff
@@ -570,24 +541,44 @@ Flash and test an image while recording serial output:
+
+_Running daft._
+
+1. Start daft by executing the following command in the PC Host: `daft DUT PATH/TO/IMG.img`. Replace DUT with the desired DUT e.g. `minnowboard` or `joule` and `PATH/TO/IMG.img` with the image you want to flash. You can also run the command directly in BBB but then it won't lock the DUTs. Running daft in BBB is done with the command `aft DUT PATH/TO/IMG.img`.In case you want to only flash or test ...
```
Add one space here:
`img.In case`

skuusela commented on README.MD:
```diff
@@ -570,24 +541,44 @@ Flash and test an image while recording serial output:
+_Running daft._
+
+1. Start daft by executing the following command in the PC Host: `daft DUT PATH/TO/IMG.img`. Replace DUT with the desired DUT e.g. `minnowboard` or `joule` and `PATH/TO/IMG.img` with the image you want to flash. You can also run the command directly in BBB but then it won't lock the DUTs. Running daft in BBB is done with the command `aft DUT PATH/TO/IMG.img`.In case you want to only flash or test ...
+2. DAFT then checks if the device is blacklisted or locked in the lockfiles. If not, then it locks it so other simultaneous DAFT runs won't use the same DUT. Or if you call DAFT explicitly e.g. daft joule1, it will only look for that device.
```
Use back ticks here?:
daft joule1 => `daft joule1`

skuusela commented on README.MD:
```diff
@@ -570,24 +541,44 @@ Flash and test an image while recording serial output:
+1. Start daft by executing the following command in the PC Host: `daft DUT PATH/TO/IMG.img`. Replace DUT with the desired DUT e.g. `minnowboard` or `joule` and `PATH/TO/IMG.img` with the image you want to flash. You can also run the command directly in BBB but then it won't lock the DUTs. Running daft in BBB is done with the command `aft DUT PATH/TO/IMG.img`.In case you want to only flash or test ...
+2. DAFT then checks if the device is blacklisted or locked in the lockfiles. If not, then it locks it so other simultaneous DAFT runs won't use the same DUT. Or if you call DAFT explicitly e.g. daft joule1, it will only look for that device.
+3. BBB reboots the DUT and proceeds to start the support image in the DUT. The support image is needed so that the target image can be flashed to the SD card or to another memory storage. BBB emulates itself as multiple components to enable automated and simple function:
+   1. As a keyboard (to get through DUT bios).
```
bios => BIOS

skuusela replied:
Maybe could also add:
`to get through DUT bios and select correct device to boot from`

skuusela commented on README.MD:
```diff
@@ -570,24 +541,44 @@ Flash and test an image while recording serial output:
+3. BBB reboots the DUT and proceeds to start the support image in the DUT. The support image is needed so that the target image can be flashed to the SD card or to another memory storage. BBB emulates itself as multiple components to enable automated and simple function:
+   1. As a keyboard (to get through DUT bios).
+   2. As a USB drive (so that the DUT can boot the support image).
+   3. As a ethernet (to give the DUT an IP address, ethernet is needed so the BBB can connect to the DUT and run commands on it required when flashing or testing).
```
Maybe more like: `As an ethernet interface`
And maybe remove the: `to give the DUT an IP address`

skuusela commented on README.MD:
```diff
@@ -570,24 +541,44 @@ Flash and test an image while recording serial output:
+   3. As a ethernet (to give the DUT an IP address, ethernet is needed so the BBB can connect to the DUT and run commands on it required when flashing or testing).
+4. When the support image boots, it gets an IP address from the BBBs DHCP server (NOT THE PC HOSTS). If you want to, you can SSH to the DUT from the PC Host by first doing a SSH to the BBB and then another SSH to the DUT.
+5. Flash the target image fetched from the PC Host via NFS to the DUTs memory storage. The DUT doesn't connect directly to the PC Host but instead through the BBB, which forwards the connection both ways.
+6. Run the tests on DUT.
```
Maybe add that it reboots and this time boots the image that was flashed.

skuusela commented on README.MD:
```diff
@@ -570,24 +541,44 @@ Flash and test an image while recording serial output:
 section. If flashing or entering test mode fails DAFT will blacklist the BBB
 used by locking the device until `/etc/daft/lockfiles/<device>` is emptied or
-removed. Also be aware that BBB has a watchdog that checks periodically if it
+removed.Also be aware that BBB has a watchdog that checks periodically if it
```
Don't remove that one space.

skuusela commented on README.MD:
```diff
@@ -615,8 +606,18 @@ bb_ip = 192.168.30.3
+
 ## 3. DAFT and AFT settings and commandline interface
 
+This section includes information about DAFT and AFT config files, CLI and alsoexample DAFT run where all the steps that DAFT makes are explained.
```
`alsoexample`

skuusela commented on README.MD:
```diff
@@ -796,8 +861,8 @@ rm am335x_evm.sh autoconfigure_usb0.sh capemgr.sh
-    'Acquire::http::Proxy "http://yourproxyaddress:proxyport";' # Add this line to the file
-apt update
+    Acquire::http::Proxy "http://yourproxyaddress:proxyport"; # Add this line to t
+apt updatehe file
```
Typo

Commit:
Update README.MD

Add 2 diagrams, which explain the BBB boot and daft runs. Reorder sections better. Also add more troubleshooting information and fix and complete some other sections.

Signed-off-by: Christian da Costa <christian.da.costa@intel.com>

skuusela commented on README.MD:
```diff
@@ -519,6 +487,7 @@ test_plan = iot_qatest
 target_device = /dev/mmcblk0
 root_partition = /dev/mmcblk0p3
 service_mode = Ubuntu
+test_mode = yocto
```
This shouldn't exist anymore.

skuusela commented on README.MD:
```diff
@@ -530,17 +499,21 @@ test_plan = iot_qatest
 target_device = /dev/mmcblk0
 root_partition = /dev/mmcblk0p3
 service_mode = Ubuntu
+test_mode = yocto
```
Same here.

skuusela commented:
Your pictures got lost also.

skuusela commented:
Accidental close....

skuusela commented on README.MD:
```diff
@@ -659,7 +662,6 @@ DAFT commandline interface options:
 * **--update**: Update AFT on the BBB filesystem and DAFT on the PC. This should
   be ran while being on the root DAFT repository directory and with root
   permission.
-* **--setout**: Flash DUT and reboot it in test mode without running test stuff
```
Don't remove this.

skuusela commented on README.MD:
```diff
@@ -711,8 +713,10 @@ these files:
 * **target_device**: Path to the target device that the image to be tested is
   flashed to.
-* **service_mode**: String included in _/proc/version_ when DUT is booted with
+* **service_mode**: String included in 'uname -a' when DUT is booted with
```
Don't change this.

skuusela commented on README.MD:
```diff
@@ -711,8 +713,10 @@ these files:
-* **service_mode**: String included in _/proc/version_ when DUT is booted with
+* **service_mode**: String included in 'uname -a' when DUT is booted with
   support image.
+* **test_mode**: String included in 'uname -a' when DUT is booted with the image
```
Shouldn't be added.
